tpm2tools/session.go

```diff
@@ -0,0 +1,92 @@
+package tpm2tools
+
+import (
+	"io"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+type session interface {
+	io.Closer
+	Auth() (tpm2.AuthCommand, error)
+}
+
+type pcrSession struct {
+	rw      io.ReadWriter
+	session tpmutil.Handle
+	sel     tpm2.PCRSelection
+}
+
+func newPCRSession(rw io.ReadWriter, sel tpm2.PCRSelection) (session, error) {
+	if len(sel.PCRs) == 0 {
+		return nullSession{}, nil
+	}
+	// This session assumes the bus is trusted, so we:
+	// - use nil for tpmkey, encrypted salt, and symmetric
+	// - use and all-zeros caller nonce, and ignore the returned nonce
+	// As we are creating a plain TPM session, we:
+	// - setup a policy session
+	// - don't bind the session to any particular key
+	session, _, err := tpm2.StartAuthSession(
+		rw,
+		/*tpmkey=*/ tpm2.HandleNull,
+		/*bindkey=*/ tpm2.HandleNull,
+		/*nonceCaller=*/ make([]byte, sessionHashAlg.Size()),
+		/*encryptedSalt=*/ nil,
+		/*sessionType=*/ tpm2.SessionPolicy,
+		/*symmetric=*/ tpm2.AlgNull,
+		/*authHash=*/ sessionHashAlgTpm)
```

Comment:
Here and the later call to StartAuthSession (line 62-65) should have the parameter name comments on the same side

Reply:
Fixed, and moved into a dedicated helper function.